core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -757,6 +679,40 @@ class TiBatchWrite(@transient val df: DataFrame,
   private def locatePhysicalTable(row: TiRow): Long = {
     tiTableInfo.getId
   }
+
+  private def splitTableRegion(wrappedRowRdd: RDD[WrappedRow]) = {
+    // when data to be inserted is too small to do region split, we check is user set region split num.
+    // If so, we do region split as user's intention. This is also useful for writing test case.
+    if (options.enableRegionSplit) {
+      if (options.regionSplitNum != 0 && isEnableSplitRegion) {
+        tiDBJDBCClient
+          .splitTableRegion(
+            options.database,
+            options.table,
+            0,
+            Int.MaxValue,
+            options.regionSplitNum
+          )
+      } else {
+        val rowSize = tiTableInfo.getEstimatedRowSizeInByte
+        val regionSplitNum = (wrappedRowRdd.count() * rowSize) / (96 * 1024 * 1024)
```

Comment:
suggest to sum the data to be inserted to calculate the real size
@birdstorm what do u think?